src/transformers/modeling_deberta.py

```diff
@@ -0,0 +1,1233 @@
+# coding=utf-8
+# Copyright 2020 Microsoft
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch DeBERTa model. """
+
+import copy
+import json
+import logging
+import math
+import os
+from collections import Sequence
+
+import torch
+from packaging import version
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from .configuration_deberta import DeBERTaConfig
+from .file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_callable
+from .modeling_outputs import BaseModelOutput, SequenceClassifierOutput
+from .modeling_utils import PreTrainedModel
+
+
+if version.Version(torch.__version__) >= version.Version("1.0.0"):
+    from torch import _softmax_backward_data as _softmax_backward_data
+else:
+    from torch import softmax_backward_data as _softmax_backward_data
+
+__all__ = [
+    "DeBERTaModel",
+    "DeBERTaForSequenceClassification",
+    "DeBERTaPreTrainedModel",
+    "DEBERTA_PRETRAINED_MODEL_ARCHIVE_LIST",
+]
```

Comment:
Is this necessary?

Reply:
By doing this, we can avoid import other components in the file, including those modules imported in this file, when we do 
from .modeling_deberta import *

Reply:
Unfortunately, I think this differs from what we do in other models and there is no good reason to have the DeBERTa model differ on that front, as it would introduce a discrepancy between this model's API and other models' APIs.